File: android/LeakCanaryDemo/app/src/main/java/com/example/yq/leakcanarydemo/SecondActivity.java
package com.example.yq.leakcanarydemo;

import android.os.Handler;
import android.os.Message;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.example.yq.leakcanarydemo.application.MyApplication;

public class SecondActivity extends AppCompatActivity {

    static A a;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        a = new A();

        findViewById(R.id.btnFinishSecond).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }

    class A{}

    @Override
    protected void onDestroy() {
        super.onDestroy();
        MyApplication.getRefWatcher().watch(this);
    }
}

File: android/ARouterDemo/app/src/main/java/com/my/arouterdemo/activity/SecondActivity.kt
package com.my.arouterdemo.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import com.alibaba.android.arouter.facade.annotation.Route
import com.my.arouterdemo.R
import com.my.common.Const
import kotlinx.android.synthetic.main.activity_second.*

@Route(path = Const.PATH_SECOND_ACT)
class SecondActivity : AppCompatActivity(), View.OnClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)

        val param = intent.getStringExtra(KEY_STR)

        Log.i(TAG, "onCreate, param: $param")

        btn_second.setOnClickListener(this)
    }


    override fun onClick(v: View?) {
        when (v) {
            btn_second -> {
                setResult(RESULT_OK)
                finish()
            }
        }
    }

    companion object {
        const val TAG = "SecondActivity"
        const val KEY_STR = "key_str"
    }

}